### DISCUSSION TITLE
Internationalization / Translation?

### DISCUSSION THREAD

**Discussion by codetalker78** [JarvyJ/HomeIntent#149] [Ideas]
Hi, 
I love the idea of having a component that takes care of the tedious rhasspy <--> home assistant integration :-)

A key feature of a voice assistant is to be able to talk to it in the user's native language. Both rasspy and home assistant already support many languages, but I have not found anything about language capabilities of HomeIntent? 
Is internationalization / translation planned for the future?

**Comment by JarvyJ** [comment]
Yes! I really want to do this. The biggest struggle for me is ensuring good translations that don't cause system confusion, are relatively consistent, and that performance remains good.

For English, I've done a lot of testing and tweaking to make sure sentences work reliably, and would want to ensure that the integrity holds for other languages. Part of it is also figuring out how internationalization would work with rhasspy. Do folks use custom wakewords or a different TTS engine? And then how do they perform on a Pi 3 (which is the lowest supported Pi version).

For an initial step in the right direction, I could always work on just getting the intent handler working with translations first, so that people could just integrate Home Intent with a [pre-existing Rhasspy setup](https://homeintent.jarvy.io/getting-started/advanced-features/external-rhasspy/).

**Reply by JarvyJ** [reply]
I've started an issue around getting i18n in (#162) that I think should work out okay. We're going for a slightly more complex approach that should allow for a lot of flexibility for translations. 

I'm planning to work on the first part of i18n for the December release.

While I wont have actual translations OOTB, the framework will be there so people can extend it. It will also be available for folks who use external Rhasspy first, as I still need to research using Rhasspy with other languages. But, it'll get there when it gets there!

**Comment by JarvyJ** [comment]
With the new release that just came out, we now have the ability to support translations! It loads rhasspy with the correct language profile and attempts to load the home Intent complements with the correct language (there aren't any yet, but folks can contribute translations now!)

More info in the docs: https://homeintent.io/reference/translations/translating-components/

**Comment by codetalker78** [comment]
Excellent! I will see how far I can get in my Christmas vacation :-)
